cds_dojson/marc21/fields/utils.py

```diff
@@ -309,39 +309,79 @@ def _get_correct_video_contributor_role(role):
     return tranlations[role.lower()]
 
 
-def _extract_json_ids(info):
+def _get_correct_books_contributor_role(role):
+    """Clean up roles."""
+    tranlations = {
+        'author': 'Author',
+        'dir.': 'Supervisor',
+        'dir': 'Supervisor',
+        'ed.': 'Editor',
+        'ed': 'Editor',
+        'ill.': 'Ilustrator',
+        'ill': 'Ilustrator',
+    }
+    return tranlations[role.lower()]
+
+
+def _extract_json_ids(info, provenence='source'):
     """Extract author IDs from MARC tags."""
     SOURCES = {
-        'AUTHOR|(INSPIRE)': 'INSPIRE',
+        'AUTHOR|(INSPIRE)': 'INSPIRE ID',
         'AUTHOR|(CDS)': 'CDS',
-        '(SzGeCERN)': 'CERN'
+        'AUTHOR|(SzGeCERN)': 'CERN'
     }
-    regex = re.compile(r'((AUTHOR\|\((CDS|INSPIRE)\))|(\(SzGeCERN\)))(.*)')
+    regex = re.compile(r'(AUTHOR\|\((CDS|INSPIRE|SzGeCERN)\))(.*)')
     ids = []
-    match = regex.match(info.get('0', ''))
-    if match:
-        ids.append({
-            'value': match.group(5),
-            'source': SOURCES[match.group(1)]
-        })
+    author_ids = force_list(info.get('0', ''))
+    for author_id in author_ids:
+        match = regex.match(author_id)
+        if match:
+            ids.append(
+                {
+                    'value': match.group(3),
+                    provenence: SOURCES[match.group(1)]
+                })
     # Try and get the IDs from the auto-completion
     try:
-        ids.append({'value': info['cernccid'], 'source': 'CERN'})
+        ids.append({'value': info['cernccid'], provenence: 'CERN'})
     except KeyError:
         pass
     try:
-        ids.append({'value': info['recid'], 'source': 'CDS'})
+        ids.append({'value': info['recid'], provenence: 'CDS'})
     except KeyError:
         pass
     try:
-        ids.append({'value': info['inspireid'], 'source': 'INSPIRE'})
+        ids.append({'value': info['inspireid'], provenence: 'INSPIRE ID'})
     except KeyError:
         pass
 
     return ids
 
 
-def build_contributor(value):
+def build_contributor_books(value):
+    """Create the contributors for books"""
+    if not value.get('a'):
+        return []
+
+    contributor = {
+        'ids': _extract_json_ids(value, 'schema') or None,
+        'full_name': value.get('name') or value.get('a'),
+        'email': value.get('email'),
```

Comment:
do you clean somehow the values? for example strip of white spaces or perform a check if the value is present? Not checking it caused problems in keywords in videos, blank fields were migrated

Reply:
I added some cleaning, let me know if you think I should be doing even more